File: src/main/java/io/junseok/mealmate/domain/member/repository/MemberRepository.java
package io.junseok.mealmate.domain.member.repository;

import io.junseok.mealmate.domain.member.entity.Member;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface MemberRepository extends JpaRepository<Member,Long> {
    boolean existsByEmail(String email);

    Optional<Member> findOneWithAuthoritiesByEmail(String email);

    Optional<Member> findByEmail(String email);
}

File: src/main/java/io/junseok/mealmate/domain/member/repository/MemberRepository.kt
package io.junseok.mealmate.domain.member.repository

import io.junseok.mealmate.domain.member.entity.Member
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface MemberRepository : JpaRepository<Member, Long> {
    fun existsByEmail(email: String): Boolean

    fun findOneWithAuthoritiesByEmail(email: String): Member?

    fun findByEmail(email: String): Member?
}
